src/app/beer_garden/router.py

```diff
@@ -18,56 +17,178 @@
 from beer_garden.errors import RoutingRequestException
 
 
-class Route_Type(Enum):
-    CREATE = "CREATE"
-    READ = "READ"
-    UPDATE = "UPDATE"
-    DELETE = "DELETE"
+def _local_garden():
+    return beer_garden.config.get("garden.name")
 
 
-def enum_lookup(enum, value):
+def enum_lookup(enum, name):
     for record in enum:
-        if record.value == value:
+        if record.name == name:
             return record
     return None
 
 
-class Route_Class(Enum):
-    COMMAND = brewtils.models.Command.schema
-    INSTANCE = brewtils.models.Instance.schema
-    JOB = brewtils.models.Job.schema
-    REQUEST = brewtils.models.Request.schema
-    REQUEST_TEMPLATE = brewtils.models.RequestTemplate.schema
-    SYSTEM = brewtils.models.System.schema
-    GARDEN = brewtils.models.Garden.schema
-    EVENT = brewtils.models.Event.schema
-    LOGGING = brewtils.models.LoggingConfig.schema
-    QUEUE = brewtils.models.Queue.schema
-
-
 class Routable_Garden_Name(Enum):
     PARENT = 1
     CHILD = 2
 
 
-Routing_Eligible = [
-    Route_Class.INSTANCE,
-    Route_Class.REQUEST,
-    Route_Class.REQUEST_TEMPLATE,
-    Route_Class.SYSTEM,
-    Route_Class.EVENT,
-]
+def forward_elgible(forward: brewtils.models.Forward):
+    """
+    Routing logic to determine if a request should be forwarded
 
+    :param forward: Forward request object that needs to be evaluated
+    :return: Booelan response on if the object should be forwarded
+    """
+    return (
+        forward.source_garden_name is not None
+        and forward.target_garden_name is not None
+        and forward.source_garden_name is not Routable_Garden_Name.CHILD.name
+        and forward.target_garden_name is not forward.source_garden_name
+        and forward.target_garden_name is not _local_garden()
+    )
 
-def _local_garden():
-    return beer_garden.config.get("garden.name")
+
+def forward_elgible_request(forward: brewtils.models.Forward):
+    """
+    Preps Request object forwards prior to evaluation
+    :param forward: Forward request object that needs to be evaluated
+    :return: Boolean response on if the object should be forwarded
+    """
+    # if forward.target_garden_name is None:
+    #     for arg in forward.args:
+    #         if isinstance(arg, (brewtils.models.Request, brewtils.models.RequestTemplate)):
+    #             forward.target_garden_name = get_system_mapping(
+    #                 name_space=arg.namespace,
+    #                 system=arg.system,
+    #                 version=arg.system_version,
+    #             )
```

Comment:
Block-commented code needs to be removed before merging.

Reply:
I can clean this up. I commented this to ensure it didn't run while testing. This should be functional after we get the plug-in registration working to make to `namespaces` and `garden_names`